src/main/java/software/amazon/encryption/s3/S3EncryptionClient.java

```diff
@@ -676,6 +683,22 @@ public Builder enableMultipartPutObject(boolean _enableMultipartPutObject) {
             return this;
         }
 
+        /**
+         * Sets the buffer size for safe authentication used when delayed authentication mode is disabled.
+         * If buffer size is not given during client configuration, default buffer size is set to 64MiB.
+         * @param bufferSize the desired buffer size in Bytes.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         * @throws S3EncryptionClientException if the specified buffer size is outside the allowed bounds
+         */
+        public Builder maxBufferSize(long bufferSize) throws IllegalArgumentException {
```

Comment:
I think we should rename this to `setBufferSize`, it makes more sense in the context of the client builder

Reply:
does this ever throw `IllegalArgumentException`?

Reply:
It's a nit, will remove `IllegalArgumentException` and clean up the code.